src/ng/compile.js

```diff
@@ -2386,8 +2412,20 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             removeControllerBindingWatches =
               initializeDirectiveBindings(controllerScope, attrs, controller.instance, bindings, controllerDirective);
           }
+
+          if (isObject(controllerDirective.require) && !isArray(controllerDirective.require)) {
+            var controllerObj = getControllers(name, controllerDirective.require, $element, elementControllers);
+            extend(controller.instance, controllerObj);
```

Comment:
Doesn't this mean that controllers whose constructors return a value won't be properly handled ?
Do we really need to set the required controllers as properties on the requiring controller ? How about just passing the object as argument to `$onInit` ?